File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int i=0,n=nums.length,j=n-1;
        int m =0;
        while( j>i){
            m=i+(j-i)/2;
            if(nums[j]>nums[m])j=m;
            else i=m+1;
        }
        return  nums[i];
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = []
        n=len(nums)
        # for i in range(n):
        #     if n == 0:
        #         left.append(nums[i])
        #     else:
        #         left.append(left[i-1]*nums[i])
        mult = 1
        zeroCount=0
        oneZero=1
        for i in nums:
            if i == 0:
                zeroCount+=1
            if zeroCount >= 2:
                return [0]*n
            if zeroCount == 1 and i==0:
                continue
            mult *= i
        for i in nums:
            if zeroCount == 1:
                if i == 0:
                    left.append(mult)
                else:
                    left.append(0)
            else:
                left.append(int(mult/i))
        return left
            

        